recipe/meta.yaml

```diff
@@ -16,7 +16,8 @@ source:
   # arrow has the unfortunate habit of changing tags of X.0.0 in the
   # lead-up until release -> don't use github sources on main
   # - url: https://github.com/apache/arrow/archive/refs/tags/apache-arrow-{{ version }}.tar.gz
-  - url: https://dist.apache.org/repos/dist/release/arrow/arrow-{{ version }}/apache-arrow-{{ version }}.tar.gz
+  - url: https://www.apache.org/dyn/closer.lua/arrow/arrow-{{ version }}/apache-arrow-{{ version }}.tar.gz?action=download
+    fn: apache-arrow-{{ version }}.tar.gz
```

Comment:
Seeing this on the `14.x` branch right now:
```
Downloading https://www.apache.org/dyn/closer.lua/arrow/arrow-14.0.2/apache-arrow-14.0.2.tar.gz?action=download
WARNING: Error: HTTP 404 NOT FOUND for url <https://www.apache.org/dyn/closer.lua/arrow/arrow-14.0.2/apache-arrow-14.0.2.tar.gz?action=download>
Elapsed: 00:00.432166
``` 

Let's hope it's intermittent. I'm not a big fan of dynamically generated links due to such failures...